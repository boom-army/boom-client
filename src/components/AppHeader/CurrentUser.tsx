import NotificationsActiveIcon from "@mui/icons-material/NotificationsActive";
import NotificationsIcon from "@mui/icons-material/Notifications";
import React from "react";
import { Badge, Box, Grid, IconButton, styled } from "@mui/material";
import { WalletMultiButton } from "../WalletMUIConnect";
import { useNewMentions, useSidebarState } from "../../hooks";

export const CurrentUser: React.FC = () => {
  const { sidebar, toggleRightSidebar } = useSidebarState();
  const { newMentions } = useNewMentions();

  const NotificationPulse = styled(NotificationsActiveIcon)(({ theme }) => ({
    padding: "0.15rem",
    animation: "pulse 5s infinite",
    borderRadius: "50%",
    border: `1px solid ${theme.palette.text.primary}`,
    borderColor: theme.palette.text.primary,
    "@keyframes pulse": {
      "0%": {
        opacity: 1,
      },
      "50%": {
        opacity: 0.5,
      },
      "100%": {
        opacity: 1,
      },
    },
  }));

  return (
    <>
      <Grid
        container
        direction="row"
        justifyContent="flex-start"
        alignItems="center"
      >
        <Box mr={1} display="flex" alignItems="center">
          <WalletMultiButton />
          {sidebar.rightNotificationsFull ? (
            <IconButton
              aria-label="notifications-open"
              onClick={() =>
                toggleRightSidebar(!sidebar.rightNotificationsFull)
              }
            >
              {newMentions?.length ? (
                <Badge badgeContent={newMentions?.length} color="primary">
                  <NotificationPulse />
                </Badge>
              ) : (
                <NotificationPulse />
              )}
            </IconButton>
          ) : (
            <IconButton
              aria-label="notifications"
              onClick={() =>
                toggleRightSidebar(!sidebar.rightNotificationsFull)
              }
            >
              {newMentions?.length ? (
                <Badge badgeContent={newMentions?.length} color="primary">
                  <NotificationsIcon />
                </Badge>
              ) : (
                <NotificationsIcon />
              )}
            </IconButton>
          )}
        </Box>
      </Grid>
    </>
  );
};
