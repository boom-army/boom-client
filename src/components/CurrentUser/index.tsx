import { Box, Grid, IconButton } from "@mui/material";
import MenuIcon from "@mui/icons-material/Menu";
import { useToggleDrawer } from "../../hooks";
import { WalletMultiButton } from "@solana/wallet-adapter-material-ui";

export const CurrentUser = () => {
  const toggleDrawer = useToggleDrawer();
  
  return (
    <>
      <Grid
        container
        direction="row"
        justifyContent="flex-start"
        alignItems="center"
      >
        <Box mr={1} display="flex" alignItems="center" >
        <WalletMultiButton />
          <IconButton
            color="primary"
            onClick={toggleDrawer(true)}
            sx={{ ml: 1 }}
          >
            <MenuIcon />
          </IconButton>
        </Box>
      </Grid>
    </>
  );
};
