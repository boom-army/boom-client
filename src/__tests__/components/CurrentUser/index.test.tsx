import React from "react";
import { CurrentUser } from "../../../components/CurrentUser/index";
import { act, fireEvent, render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("<CurrentUser/> component :", () => {
  test("display <CurrentUser/> component ", async () => {
    let rendered = render(<CurrentUser />, { wrapper: MemoryRouter });
    expect(rendered).toMatchSnapshot();
  });
});

test("Assertion testing of  <CurrentUser/> component ", async () => {
  render(<CurrentUser />, { wrapper: MemoryRouter });
  expect(screen.getByRole("link", { name: "Boom Î²" })).toHaveAttribute(
    "href",
    "/"
  );
});
