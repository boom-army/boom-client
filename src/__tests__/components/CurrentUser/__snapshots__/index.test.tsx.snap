// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CurrentUser/> component : display <CurrentUser/> component  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .css-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.css-1 {
  margin-right: 8px;
}

.css-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
}

<body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container css-0"
      >
        <div
          class="MuiBox-root css-1"
        >
          <a
            href="/"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-2"
            >
              Boom β
            </h6>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": .css-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.css-1 {
  margin-right: 8px;
}

.css-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
}

<div>
    <div
      class="MuiGrid-root MuiGrid-container css-0"
    >
      <div
        class="MuiBox-root css-1"
      >
        <a
          href="/"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2"
          >
            Boom β
          </h6>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
